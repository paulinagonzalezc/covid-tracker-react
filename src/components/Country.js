import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

const Country = (props) => {
  const { name, flag, cases } = props;
  return (
    <div className="country-container">
      <div className="detail-title">
        <Link to={`details/${name}`} data-testid="detail-link"><ArrowForwardIcon className="arrow" /></Link>
        <h2 className="country-title" data-testid={name}>{name}</h2>
      </div>
      <img className="country-flag" alt="flag" src={flag} />
      <p className="country-cases">
        Total Cases:
        {' '}
        {cases}
      </p>
    </div>
  );
};

Country.propTypes = {
  name: PropTypes.string.isRequired,
  cases: PropTypes.number.isRequired,
  flag: PropTypes.string.isRequired,
};

export default Country;
