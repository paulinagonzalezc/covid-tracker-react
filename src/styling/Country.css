.country-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #35548b;
  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
  border-radius: 5px;
}

.country-container:nth-child(even) {
  background-color: #4063a7;
}

.country-flag {
  width: 190px;
  height: 120px;
}

.detail-title {
  display: flex;
  flex-wrap: wrap;
  margin: 10px 0;
}

.arrow {
  height: 20px;
  padding: 0;
  margin: 0;
  color: white;
}

.arrow:hover {
  background-color: gray;
  border-radius: 10px;
}

.country-title {
  margin: 0;
  font-size: 18px;
  color: white;
  font-weight: 300;
}

.country-cases {
  color: white;
}
