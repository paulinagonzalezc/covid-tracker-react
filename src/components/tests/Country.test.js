import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Country from '../Country';

describe('Country component test', () => {
  test('should render', () => {
    const country = render(
      <Provider store={store}>
        <BrowserRouter>
          <Country
            key={1}
            id={1}
            name="Mexico"
            flag="https://disease.sh/assets/img/flags/mx.png"
            cases={2800}
          />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(country).toMatchSnapshot();
    const title = screen.getByText(/Mexico/);
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('country-title');
    expect(screen.getByTestId('detail-link')).toBeInTheDocument();
  });
});
