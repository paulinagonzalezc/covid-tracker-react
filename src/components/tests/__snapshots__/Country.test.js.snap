// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="country-container"
      >
        <div
          class="detail-title"
        >
          <a
            data-testid="detail-link"
            href="/details/Mexico"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </a>
          <h2
            class="country-title"
            data-testid="Mexico"
          >
            Mexico
          </h2>
        </div>
        <img
          alt="flag"
          class="country-flag"
          src="https://disease.sh/assets/img/flags/mx.png"
        />
        <p
          class="country-cases"
        >
          Total Cases:
           
          2800
        </p>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="country-container"
    >
      <div
        class="detail-title"
      >
        <a
          data-testid="detail-link"
          href="/details/Mexico"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </a>
        <h2
          class="country-title"
          data-testid="Mexico"
        >
          Mexico
        </h2>
      </div>
      <img
        alt="flag"
        class="country-flag"
        src="https://disease.sh/assets/img/flags/mx.png"
      />
      <p
        class="country-cases"
      >
        Total Cases:
         
        2800
      </p>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
